pkg/provider/azure_loadbalancer.go

```diff
@@ -1384,6 +1385,7 @@ func (az *Cloud) reconcileLoadBalancer(clusterName string, service *v1.Service,
 		}
 		if changed {
 			dirtyLb = true
+			_ = az.lbCache.Delete(*lb.Name)
```

Comment:
The LB is not updated yet here (it would be updated after `CreateOrUpdateLB()` is called), hence caches shouldn't be refreshed here and below.